### PR TITLE
Document new feature StdoutErrConsumer [9750]

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
This pull request documents changes in eProsima/Fast-DDS#1340.

* Update Makefile so the spelling errors are shown in the terminal.
* New CMake option `LOG_CONSUMER_DEFAULT`.
* New logging module classes and hierarchy.
* New StdoutErrConsumer.
* How to configure new consumer in XML file

Signed-off-by: JLBuenoLopez-eProsima <joseluisbueno@eprosima.com>

### CODE REVIEW

File: docs/fastdds/logging/logging_module.rst [hunk 1]
```diff
@@ -24,18 +24,26 @@ The logging module provides the following classes:
...
+  It includes the member functions that derived classes should overload to consume log entries.
 
-  - |StdoutConsumer-api| derives from |LogConsumer-api|.
-    It defines how to consume log entries for outputting to STDOUT (see :ref:`dds_layer_log_consumer_stdout`).
-  - |FileConsumer-api| derives from |LogConsumer-api|.
-    It defines how to consume log entries for outputting to a user specified file (see
-    :ref:`dds_layer_log_consumer_file`).
+  - |OstreamConsumer-api| derives from |LogConsumer-api|.
+    It defines how to consume log entries for outputting to an `ostream` object.
+    It includes a member function that derived classes chould overload to define the desired `ostream` object.
```
Comment (EduPonz) [inline docs/fastdds/logging/logging_module.rst:31]:
```suggestion
    It includes a member function that derived classes must overload to define the desired `ostream` object.
```

File: docs/fastdds/xml_configuration/log.rst [hunk 2]
```diff
@@ -45,30 +45,45 @@ ConsumerDataType
...
 | ``<class>``       | The class of the consumer.                                               | ``StdoutConsumer``    |
 |                   |                                                                          +-----------------------+
+|                   |                                                                          | ``StdoutErrConsumer`` |
+|                   |                                                                          +-----------------------+
 |                   |                                                                          | ``FileConsumer``      |
 +-------------------+--------------------------------------------------------------------------+-----------------------+
-| ``<property>``    | This element is used to configured the file consumer and therefore |br|  | :ref:`xmllogprop`     |
-|                   | only applies if ``<class>`` is set to ``FileConsumer``.                  |                       |
+| ``<property>``    | This element is used to configured the log consumer and only applies|br| | :ref:`xmllogprop`     |
```
Comment (EduPonz) [inline docs/fastdds/xml_configuration/log.rst:52]:
```suggestion
| ``<property>``    | This element is used to configure the log consumer and only applies|br| | :ref:`xmllogprop`     |
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/636/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/637/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/638/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/639/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/640/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/641/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/642/)

File: docs/fastdds/logging/logging_configuration.rst [hunk 3]
```diff
@@ -187,7 +187,7 @@ The logging module's configuration can be reset to default settings with member
     * Disabling :ref:`dds_layer_log_file_context` component.
     * Enabling :ref:`dds_layer_log_function_name` component.
     * Clear all :ref:`dds_layer_log_filter`.
-    * Clear all consumers and set a STDOUT consumer.
+    * Clear all consumers and set a STDOUT or STDOUTERR consumer (depending on the CMake option |LOG_CONSUMER_DEFAULT|).
```
Comment (EduPonz) [inline docs/fastdds/logging/logging_configuration.rst:190]: What about the threshold level? I'm guessing the new `STDOUTERR` will have the default value, right?

File: docs/fastdds/logging/logging_filtering.rst [hunk 4]
```diff
@@ -108,4 +108,4 @@ The logging module's filters can be reset with member function |Log::Reset-api|.
     * Disabling :ref:`dds_layer_log_file_context` component.
     * Enabling :ref:`dds_layer_log_function_name` component.
     * Clear all :ref:`dds_layer_log_filter`.
-    * Clear all consumers.
+    * Clear all consumers and set a STDOUT or STDOUTERR consumer (depending on the CMake option |LOG_CONSUMER_DEFAULT|).
```
Comment (EduPonz) [inline docs/fastdds/logging/logging_filtering.rst:111]: I'm wondering it this was even true to begin with

File: docs/fastdds/logging/consumer.rst [hunk 5]
```diff
@@ -8,20 +8,22 @@ Consumers
...
 StdoutConsumer
 ^^^^^^^^^^^^^^
 
-|StdoutConsumer-api| is the default log consumer.
-It outputs log entries to STDOUT stream following the convection specified in :ref:`dds_layer_log_logging_spec`.
-By default, the logging module only has the StdoutConsumer.
+|StdoutConsumer-api| outputs log entries to STDOUT stream following the convection specified in
+:ref:`dds_layer_log_logging_spec`.
+It is the default and only log consumer of the logging module if the CMake option |LOG_CONSUMER_DEFAULT| is set to
+``AUTO`` or ``STDOUT``.
```
Comment (EduPonz) [inline docs/fastdds/logging/consumer.rst:26]: Or not set at all

File: docs/fastdds/xml_configuration/log.rst [hunk 6]
```diff
@@ -27,13 +27,13 @@ These are described in the following table.
 +-------------------+---------------------------------------------------------------+-----------------------+----------+
 
 The following constitutes an example of an XML configuration file that sets the |Log-api| to use one
-|StdoutConsumer-api| and one |FileConsumer-api|:
+|StdoutConsumer-api|, one |StdoutErrConsumer-api| and one |FileConsumer-api|:
```
Comment (EduPonz) [inline docs/fastdds/xml_configuration/log.rst:30]:
```suggestion
|StdoutConsumer-api|, one |StdoutErrConsumer-api|, and one |FileConsumer-api|:
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: docs/fastdds/logging/logging_configuration.rst (hunk 3 above)
Comment (JLBuenoLopez-eProsima) [reply docs/fastdds/logging/logging_configuration.rst]: Yes, the default consumer is reset.

File: docs/fastdds/logging/logging_filtering.rst (hunk 4 above)
Comment (JLBuenoLopez-eProsima) [reply docs/fastdds/logging/logging_filtering.rst]: In order to clear the filters, `Log::Reset()` must be called and this method performs all those operations.

File: docs/fastdds/xml_configuration/log.rst (hunk 6 above)
Comment (JLBuenoLopez-eProsima) [reply docs/fastdds/xml_configuration/log.rst]: Done in 2b7b8953a6a91a7f1e80fe972b8de90d78beffa0

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/644/)

File: docs/fastdds/logging/consumer.rst (hunk 5 above)
Comment (JLBuenoLopez-eProsima) [reply docs/fastdds/logging/consumer.rst]: Done in 242f3bf33764025c8523d3cfa206dfa4c22a42f6

**Review by EduPonz — DISMISSED** [review]

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/645/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/646/)

**Review by EduPonz — APPROVED** [review]

**Pull request merged**